distribution/lib/Standard/Base/0.0.0-dev/src/Data/Enso_Cloud/Enso_File.enso

```diff
@@ -344,15 +349,16 @@ Enso_Asset_Type.from (that:Text) = case that of
 File_Format_Metadata.from (that:Enso_File) = File_Format_Metadata.Value Nothing that.name (that.extension.catch _->Nothing)
 
 ## PRIVATE
+   Fetches the basic information about a file from the Cloud endpoint.
 get_file_description file:Enso_File -> JS_Object =
-    auth_header = Utils.authorization_header
-    response = HTTP.fetch file.internal_uri HTTP_Method.Get [auth_header]
-    response.if_not_error <|
-        js_object = response.decode_as_json
-        js_object.get "file" if_missing=(Error.throw (Illegal_State.Error "Invalid JSON for an Enso_File (missing `file` field): "+js_object.to_text))
+    Utils.http_request_as_json HTTP_Method.Get file.internal_uri
 
 ## PRIVATE
-get_s3_url_for_file file:Enso_File -> Text =
+   The returned URL may be valid only for a very short time, so it should not be
+   stored anywhere.
+get_download_url_for_file file:Enso_File -> Text =
     file_description = get_file_description file
-    path = file_description.get "path" if_missing=(Error.throw (Illegal_State.Error "Invalid JSON for an Enso_File (missing `path`): "+file_description.to_text))
+    ## TODO migrate to the pre-signed URLs once that is implemented
+       presigned_url = file_description |> get_required_field "url"
+    path = file_description |> get_required_field "file" |> get_required_field "path"
     path.replace "s3://production-enso-organizations-files/" "https://production-enso-organizations-files.s3.eu-west-1.amazonaws.com/"
```

Comment:
Pre-signed URLs are not enabled yet as the feature was apparently rolled-back.

They can be enabled and will then be tested once it is available again, possibly in another PR.